lib/src/lifecycle_module.dart

```diff
@@ -127,10 +138,84 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
     }.forEach(_logLifecycleEvents);
   }
 
+  /// If this module is in a transition state, this is the Span capturing the
+  /// transition state.
+  Span get activeSpan => _activeSpan;
+
+  /// Set internally by this module for the load span so it can be used as a
+  /// `Reference` to other spans after the span is finished.
+  SpanContext _loadContext;
+
+  /// Set internally by the parent module if this module is called by [loadChildModule]
+  SpanContext _childOfContext;
+
+  /// Builds a span that conditionally applies a followsFrom reference if this module
+  /// was loaded by a parent module.
+  ///
+  /// Returns `null` if no globalTracer is configured.
+  Span _startTransitionSpan(String operationName) {
+    final tracer = globalTracer();
+    if (tracer == null) {
+      return null;
+    }
+
+    List<Reference> references = [];
+    if (_childOfContext != null) {
+      references.add(new Reference.followsFrom(_childOfContext));
+    }
+
+    return tracer.startSpan(
+      operationName,
+      references: references,
+      tags: _defaultTags,
+    );
+  }
+
+  /// Creates a span from the start of [load] until now.
+  ///
+  /// This is different from [onLoad] as this indicates that whichever asynchronous
+  /// processes required to make this module interactible (such as any data needed
+  /// to populate its store) will be finished by this point, whereas it might
+  /// only be started in [onLoad].
+  ///
+  /// Any [tags] specified will be added to the span which results from this call.
+  @protected
+  void specifyStartupTiming(
+    StartupTimingSpecifier specifier, {
+    Map<String, dynamic> tags: const {},
+  }) {
+    // Load didn't start
+    if (_loadContext == null || _startLoadTime == null) {
+      return null;
+    }
+
+    final tracer = globalTracer();
+    if (tracer == null) {
+      return null;
+    }
+
+    tracer
+        .startSpan(
+          specifier.name,
+          references: [tracer.followsFrom(_loadContext)],
+          startTime: _startLoadTime,
```

Comment:
Would be nice if we started this span at the same time as load as opposed to passing in the start time so that we could tie spans into browser marks/measures

Reply:
Then we would end up with unfinished spans for all modules that don't implement closing these.

Reply:
Good point, didn't think about that...

...Is it a problem to have unfinished spans?

Reply:
Each unfinished span leaks a `Completer` in the span object and a `whenFinished` stream listener in the tracer (App Intelligence for us). A few might go unnoticed, but I'm not comfortable with the possibility that this will leak memory over time if we're automatically starting them for every module and only finishing them for the handful that actually implement stopping these.